pkg/kubelet/prober/prober.go

```diff
@@ -115,9 +115,21 @@ func (pb *prober) probe(ctx context.Context, probeType probeType, pod *v1.Pod, s
 		pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, events.ContainerProbeWarning, "%s probe warning: %s", probeType, output)
 		klog.V(3).InfoS("Probe succeeded with a warning", "probeType", probeType, "pod", klog.KObj(pod), "podUID", pod.UID, "containerName", container.Name, "output", output)
 	} else {
+		switch result {
+		case probe.Success:
+			pb.recordContainerEvent(pod, &container, v1.EventTypeNormal, events.ContainerHealthy, "%s probe succeeded", probeType)
```

Comment:
Emitting an event when a probe has succeeded will be very noisy since we will always have at least one event emitted per probe. Only emitting an event when something goes wrong or changes would be more meaningful to the users since that is something they might want to be aware of.